pkg/types/types.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2020 Red Hat, Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+const (
+	Hugepage1GiRequestPath = "hugepages_1Gi_request"
```

Comment:
I believe these paths are used to create the downward API file. Are these paths standardized in the k8 community somewhere? Should we be pulling in these paths from somewhere instead of defining them ourselves?

Reply:
@martinkennelly 
I can't find a standard path name in the Kubernetes code, which is why I added a `pkg/types/` so the path name could be reused in app-netutil. I even looked for others, like `cpu_limit` and `mem_request`. We don't even use the standard path of `/etc/podinfo/`. We use `/etc/podnetinfo/` to avoid any conflicts.

I used https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/ as my naming convention. Digging though the hugepage patch more (https://github.com/kubernetes/kubernetes/pull/86102/files), in  pkg/apis/core/validation/validation_test.go (was hidden because diff was too large), I see the test code uses `hugepages_request` and `hugepages_limit`. Kubernetes doesn't support mixing 1G and 2M hugepages, so should I just move to that convention? Or be forward looking and leave as is in case it is ever supported?

Reply:
Whatever naming convention I use, I could move the constants somewhere else, like `k8snetworkplumbingwg/network-attachment-definition-client`? May slow this patch down slightly.

Reply:
Latest commit change to use `hugepages_request` and `hugepages_limit`.

Reply:
I believe you can have both 1G and 2M hugepages for a pod ( https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/#api )

Reply:
@martinkennelly OK, I'll change to allow both. Any other concerns?

Reply:
_IF_ I am right about being able to use both 2m/1g hugepages at the same time, then there is a problem here. You've no way for an app to distinguish between both first off, but even then, only one will be selected I guess and populate hugepages_request/hugepages_limit. I must test this out. I haven't found docs to say I can use both yet.

Reply:
Year and half ago, my server was setup with both 1G and 2M hugpages and K8s wouldn't deploy. I found somewhere that indicated it wasn't supported. But a lot has change since then.

Reply:
I believe you can with this feature gate: --feature-gates=HugePageStorageMediumSize=true

Reply:
OK, I'll add _1G and _2M back to the "Path"  name.